### ISSUE TITLE
How do we get payment information for the sender of the Moment

### ISSUE THREAD

**Issue by jnystrom** [KL-Moment/bot#5]

**Comment by benjamincharity** [comment]
Are we:

- a) Trying to use our existing payment system
- b) Trying to use Facebook Canvas payments
- c) No current opinion

**Comment by garrettlangley** [comment]
My hope is that is resembles our email process in flow, wherein a user is eventually prompted (if no payment is on file) to submit payment. They are taken to a webpage (like 1800Flowers flow) wherein they enter payment on a page build/hosted by Moment

**Comment by benjamincharity** [comment]
Any specific reason we want to take the user out of the experience?

**Comment by garrettlangley** [comment]
1 - No. Ideally, like an Apple Pay experience, the user just taps a button or does something simple.

2 - I dont think we can securely capture the credit card information in the message (please type in your credit card number) as this exposes us to PCI certification

**Comment by benjamincharity** [comment]
1 - This doesn't seem like it would scale across the platforms/devices that FB gives us access too.

2 - You've verified canvas payments require us to consume the cc number?

**Comment by garrettlangley** [comment]
@benjamincharity no, i havent dug into Canvas Payments (not familiar at all actually) -- I, perhaps, wrongly assumed we needed to pull people out as thats how 1800Flowers and Spring did it

**Comment by jnystrom** [comment]
I would think pulling them out would be the easiest flow, since we already have a page to capture this info.  I do think we can all agree this is NOT ideal, but we have it built, so should be the easiest v1.

**Comment by benjamincharity** [comment]
Ok, I haven't looked into it enough. Didn't want to waste time digging if you had already done so. I'll look into it to verify

**Comment by benjamincharity** [comment]
OK, fwiw using canvas is basically FBs version of apple pay. 

- Users can have multiple payment types tied to their account.
- When a user goes to pay, a modal is popped (much like stripe checkout) where they can choose a payment method or add one.
- CC info never comes to us (like stripe)
- Supports static and dynamic pricing

**Comment by garrettlangley** [comment]
@benjamincharity interesting, so does it work like Apple Pay where we still get to save and re-use the card across platforms or does it only work inside Facebook?

either way, seems like capturing from our own webpage is the right v1 decision
